src/mrack/providers/podman.py

```diff
@@ -58,17 +58,42 @@ async def init(self, container_images, default_network, ssh_key, container_optio
         logger.info(f"{self.dsp_name}: Init duration {login_duration}")
 
     async def prepare_provisioning(self, reqs):
-        """Pull missing images to prepare provisioning for podman."""
+        """Prepare podman network and pull missing images to prepare provisioning."""
         image_list = set()
+        network_list = set()
 
         for req in reqs:
+            # First create list of podman network(s) to be created
+            # podman network(s) should be always created no matter what
+            network = req.get(
+                "network", f"{self.default_network}-{req['domain'].replace('.', '-')}"
+            )
+            # store the used network per requirement to be later accessed
+            req["network"] = network
+            if not await self.podman.network_exists(network):
```

Comment:
`self.podman.network_exists(network)` check is run for all reqs no matter what in sequence. So e.g. for 5 machines with the same network, it is run 4 times unnecessarily. 

Better logic would be:
- generate the list of network names in `network_list` (and as it is a set, it will have only unique)
- run `self.podman.network_exists(network):` for each network in network_list
- create the ones which doesn't exist

Reply:
Correct, will update.

Reply:
Done